src/lib/util/RunOutputProcessor.ts

```diff
@@ -76,6 +79,16 @@ export class RunOutputProcessor {
 		return this.opts.violationsCauseException ? INTERNAL_ERROR_CODE : 1;
 	}
 
+	// determines if -s flag should cause an error 
+	private shouldErrorForSeverity(minSev: number, severityForError): boolean {
+		if (severityForError == undefined) return false; // flag not used
+		if (minSev == 0) return false;
+		// We want to throw an error when the highest severity (smallest num) is 
+		// equal to or more severe (equal to or less than number-wise) than the inputted number
+		if (minSev <= this.opts.severityForError) return true; 
+		return false;
+	}
+
```

Comment:
The logic for determining if the scanner should throw and error based on the severities found and the number the user gave to flag